adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -401,11 +401,17 @@ func (h *hnsw) reassignNeighborsOf(deleteList helpers.AllowList, breakCleanUpTom
 	defer h.resetLock.Unlock()
 
 	g, ctx := enterrors.NewErrorGroupWithContextWrapper(h.logger, h.shutdownCtx)
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
I think that you can add `defer cancel()` here, bc we always eventually cancel when we exit this method